### PR TITLE
feat: auto-synced google calendar

### PR DESCRIPTION
**Pull request by transmissions11**
closes https://github.com/DormSoup/dormsoup/issues/7

### CODE REVIEW

**Comment by transmissions11** [comment]
TODO:
1. Testing
2. Database migration for `gcalId` event field(?)

File: src/gcal.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+      start: {
+        dateTime: event.date.toISOString(),
+        timeZone: "America/New_York"
+      },
+      end: {
+        // Just assume it ends 1 hour after it starts, for now.
+        dateTime: new Date(event.date.getTime() + 60 * 60 * 1000).toISOString(),
+        timeZone: "America/New_York"
+      }
+    };
```
Comment (lad6558) [inline src/gcal.ts:92]: Be careful about the date, we store the event time as if they are ISO when in fact they are in EST. You'll probably encounter that in testing

**Review by lad6558 — COMMENTED** [review]
Mostly looks good to me. Be careful about the date conversation. Otherwise, feel free to give it a test

File: src/gcal.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      start: {
+        dateTime: removeUTCMarker(event.date),
+        timeZone: "America/New_York"
+      },
+      end: {
+        // Just assume it ends 1 hour after it starts, for now.
+        dateTime: removeUTCMarker(new Date(event.date.getTime() + 60 * 60 * 1000)),
+        timeZone: "America/New_York"
+      }
+    };
```
Comment (lad6558) [inline src/gcal.ts:97]: event should have a `duration` field you can use if you want

**Review by lad6558 — APPROVED** [review]
LGTM. There is a `duration` field you can use for events. Feel free to go ahead and merge and test it on the server. You can `git pull` and `systemctl restart dormsoup.service` to test

**Pull request merged**